File: CivitasJava/civitas/CasillaImpuesto.java
package civitas;

import java.util.ArrayList;

public class CasillaImpuesto extends Casilla {
    private float importe;
    
    CasillaImpuesto (float cantidad, String s){
        super(s);
        importe = cantidad;
    }
    
    @Override
    void recibeJugador (int iactual, ArrayList<Jugador> todos){
        if (jugadorCorrecto(iactual,todos)){
            informe(iactual,todos);
            todos.get(iactual).pagaImpuesto(importe);
        }
    }
}

File: CivitasRuby/casilla_impuesto.rb
require_relative 'casilla'

module Civitas
class CasillaImpuesto < Casilla
  def initialize(importe, nombre)
    super(nombre)
    @importe = importe
  end
  
  def recibe_jugador(actual, todos)
    if jugador_correcto(actual, todos)
      informe(actual, todos)
      todos[actual].paga_impuesto(@importe)
    end
  end
  
end
end
